common/go-utils/src/inv.rs

```diff
@@ -118,33 +105,58 @@ impl FromStr for Arch {
     }
 }
 
-#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
-#[serde(rename_all = "lowercase")]
-pub enum Os {
-    Linux,
+#[derive(thiserror::Error, Debug)]
+pub enum ReadInventoryError {
+    #[error("Couldn't read Go artifact inventory.toml: {0}")]
+    Io(#[from] std::io::Error),
+    #[error("Couldn't parse Go artifact inventory.toml: {0}")]
+    Parse(#[from] toml::de::Error),
 }
 
-impl Display for Os {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        match self {
-            Os::Linux => write!(f, "linux"),
+impl<V> Inventory<V>
```

Comment:
I've moved the `resolve` function out of the default inv implementation, and added unit tests to exercise it (in the next PR: https://github.com/heroku/buildpacks-go/pull/245/files#diff-68036804ced205f96cbba06a330914b8a3b619742bfda4169424a7b91ad177aaR217-R252 :-))

Not sure there's a lot to test in the `read` function? But it might also just get eliminated altogether, it doesn't add much value as is tbh.